test/test.js

```diff
@@ -1,8 +1,164 @@
-import assert from 'assert';
-import { describe, it } from 'mocha';
+import { expect } from 'chai';
+import { describe, it, beforeEach, afterEach } from 'mocha';
+import {
+  InteractionResponseType,
+  InteractionType,
+  InteractionResponseFlags,
+} from 'discord-interactions';
+import { AWW_COMMAND, INVITE_COMMAND } from '../src/commands.js';
+import sinon from 'sinon';
+import server from '../src/server.js';
+import reddit from '../src/reddit.js';
 
-describe('awwbot', () => {
-  it('should have tests', () => {
-    assert.ok(true);
+describe('Server', () => {
+  describe('GET /', () => {
+    it('should return a greeting message with the Discord application ID', async () => {
+      const request = {
+        method: 'GET',
+        url: new URL('/', 'http://discordo.example'),
+      };
+      const env = { DISCORD_APPLICATION_ID: '123456789' };
+
+      const response = await server.fetch(request, env);
+      const body = await response.text();
+
+      expect(body).to.equal('👋 123456789');
+    });
+  });
+
+  describe('POST /', () => {
+    let verifyDiscordRequestStub;
+    let getCuteUrlStub;
+
+    beforeEach(() => {
+      verifyDiscordRequestStub = sinon.stub(server, 'verifyDiscordRequest');
+      getCuteUrlStub = sinon.stub(reddit, 'getCuteUrl');
+    });
+
+    afterEach(() => {
+      verifyDiscordRequestStub.restore();
+      getCuteUrlStub.restore();
+    });
+
+    it('should handle a PING interaction', async () => {
+      const interaction = {
+        type: InteractionType.PING,
+      };
+
+      const request = {
+        method: 'POST',
+        url: new URL('/', 'http://discordo.example'),
+      };
+
+      const env = {};
+
+      verifyDiscordRequestStub.resolves({
```

Comment:
Feels like we could stub this once at the top and be done with it?

Reply:
It feels noisy but each interaction is different so each stub resolves a different scenario in this case.